airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -63,23 +66,72 @@ public class TemporalClient {
    */
   private static final int DELAY_BETWEEN_QUERY_MS = 10;
 
-  private static final int MAXIMUM_SEARCH_PAGE_SIZE = 50;
+  public static TemporalClient get(final Configs configs) {
```

Comment:
Something we've been doing in preparation for adding Dependency Injection into Airbyte OSS is to change our various class initialization functions to take in fully-constructed dependencies when being constructed, rather than having the initialization function construct the dependencies itself.

I think that same strategy could be useful here - instead of adding a bunch of logic to the OSS TemporalClient to decide whether it needs to initialize a "cloud" or an "airbyte" WorkflowClient, this `get()` function could be rewritten to take in a `WorkflowClient`, and the logic to initialize the "airbyte" WorkflowClient can be added to a factory somewhere that gets called by ServerApp first, and then ServerApp passes that workflow client into this TemporalClient.get() function.

With this approach, all of this Temporal Cloud-specific logic can be added just to `airbyte-cloud`, including all of the Temporal Cloud environment variables that are being added in this PR. This would be nice because it would leave the OSS code less complicated, and it wouldn't allow an OSS user to accidentally set `TEMPORAL_CLOUD_ENABLED` to `true` and break their OSS deployment.

Some of the changes in this PR would probably still need to stay, like the `dangerouslyTerminateWorkflow()`, `getAllRunningWorkflows()` methods, and the changes to the `getTemporalClientWhenConnected()` method. But that is okay because those may be useful for both Cloud and OSS deployments.

What do you think?

Reply:
Of course, this is assuming that we don't want OSS users to be able to use Temporal Cloud. If that assumption is incorrect, then we may still need to keep the environment variables you've added, but it would probably still be a good idea to make this dependency-injection friendly.

Reply:
not sure if we've already moved this to cloud - I would prefer us leaving this in OSS. I think it's nice to easily allow our users to make use of Temporal Cloud if they want to.

Reply:
@davinchia yeah I had the same thought so we left it open for that option. I just refactored things to make the separation of concerns between TemporalClient and TemporalUtils much cleaner.